pandas/tests/frame/test_reductions.py

```diff
@@ -1025,6 +1025,28 @@ def test_idxmax_mixed_dtype(self):
         expected = Series([0, 2, 1, 2], index=[1, 2, 3, 4])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "func_name, expected_value",
+        [("idxmax", [0, 4]), ("idxmin", [0, 5])],
+    )
+    def test_idxmax_idxmin_convert_dtypes(self, func_name, expected_value):
+        df = DataFrame(
+            {
+                "ID": [100, 100, 100, 200, 200, 200],
+                "value": [0, 0, 0, 1, 2, 0],
+            }
+        )
+
+        df = df.convert_dtypes().groupby("ID")
```

Comment:
Instead of using `convert_dtypes`, better to just specify `dtype="Int64"` in DataFrame construction.

Reply:
I'd just like to confirm on this point - in the discussion I had with Marco in issue #40346, I understood the spirit of this new check as to replicate the conditions under which the original bug was found (in using `convert_dtypes` before `idxmax`) to make sure the bug was fixed. Would it not be preferable, in that case, to keep using the method which caused the original bug?

Reply:
Hi @rhshadrach , I'd just like to follow-up on the above - please let me know, and would be happy to push through all the updates together

Reply:
Thanks for the reminder here, sorry it fell off my radar. From https://github.com/pandas-dev/pandas/issues/40346#issuecomment-813320245 I believe @MarcoGorelli is just looking to create a DataFrame with nullable integers. My suggested change accomplishes this, but with using less of the pandas API. In general, it's desirable for tests to use as little of the API (besides what they are testing!) as possible.

Reply:
I see, thanks @rhshadrach! I'll push the changes through